executor/src/task.rs

```diff
@@ -287,6 +296,40 @@ pub async fn run_task(task: TaskCall, js: crate::JsCallback) -> Result<TaskRespo
                         req.resume(success)
                     }
                 },
+
+                RuntimeHostVm::LogEmit(req) => {
+                    {
+                        let info = req.info();
+                        let log = match info {
+                            LogEmitInfo::Num(v) => LogInfo {
```

Comment:
read abt rust recently, maybe we can do this, so it's cleaner?
```rust
LogEmitInfo::Num(v) | LogEmitInfo::Utf8(v) | LogEmitInfo::Hex(v) => LogInfo {
    message: v.to_string(),
    level: None,
    target: None,
},
```

Reply:
```
    Checking chopsticks-executor v0.9.0 (/Users/xiliangchen/projects/acala/chopsticks/executor)
error[E0308]: mismatched types
   --> src/task.rs:304:48
    |
303 |                         let log = match info {
    |                                         ---- this expression has type `LogEmitInfo<'_>`
304 |                             LogEmitInfo::Num(v) | LogEmitInfo::Utf8(v) | LogEmitInfo::Hex(v) => LogInfo {
    |                                              -                      ^ expected `u64`, found `LogEmitInfoStr<'_>`
    |                                              |
    |                                              first introduced with type `u64` here
    |
    = note: in the same arm, a binding must have the same type in all alternatives

error[E0308]: mismatched types
   --> src/task.rs:304:70
    |
303 |                         let log = match info {
    |                                         ---- this expression has type `LogEmitInfo<'_>`
304 |                             LogEmitInfo::Num(v) | LogEmitInfo::Utf8(v) | LogEmitInfo::Hex(v) => LogInfo {
    |                                              - first introduced with type `u64` here      ^ expected `u64`, found `LogEmitInfoHex<'_>`
    |
    = note: in the same arm, a binding must have the same type in all alternatives

For more information about this error, try `rustc --explain E0308`.
error: could not compile `chopsticks-executor` (lib) due to 2 previous errors
```

This is the error you will get.

Basically the `v` are different types and `TypeA | TypeB | TypeC` is not a thing in Rust